apps/next15-demo/src/app/components/BridgeHandler.tsx

```diff
@@ -0,0 +1,33 @@
+"use client";
+
+import { resolveRelations } from "@/lib/getStory";
+import { StoryblokBridgeV2 } from "@storyblok/js";
+import { useEffect } from "react";
+import { useSWRConfig } from "swr";
+
+declare global {
+  interface window {
+    StoryblokBridge: StoryblokBridgeV2;
+  }
+}
+
+export const BridgeHandler = () => {
+  const { mutate } = useSWRConfig();
+
+  useEffect(() => {
+    const bridge = new window.StoryblokBridge({
+      resolveRelations: resolveRelations,
+    });
+
+    bridge.on("input", (event) => {
+      if (!event || !event.story) return;
+      mutate(`cdn/stories/${event.story.slug}`, event.story, {
+        revalidate: false,
+      });
+    });
```

Comment:
The SWR cache is keyed by slug which is globally unique within the space. When a live update occurs we mutate the cache which triggers a re-render with the updated data. This occurs on the client only.